From: Stefan Metzmacher <metze@samba.org>
Date: Wed, 14 Sep 2022 05:25:46 +0200
Subject: [PATCH] cifs: don't send down the destination address to sendmsg for
 a SOCK_STREAM
Git-commit: 17d3df38dc5f4cec9b0ac6eb79c1859b6e2693a4
References: bsc#1193629
Patch-mainline: v6.0-rc6

This is ignored anyway by the tcp layer.

Signed-off-by: Stefan Metzmacher <metze@samba.org>
Cc: stable@vger.kernel.org
Reviewed-by: Ronnie Sahlberg <lsahlber@redhat.com>
Reviewed-by: Paulo Alcantara (SUSE) <pc@cjr.nz>
Signed-off-by: Steve French <stfrench@microsoft.com>
Acked-by: Paulo Alcantara <palcantara@suse.de>
---
 fs/cifs/transport.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/fs/cifs/transport.c b/fs/cifs/transport.c
index c2fe035e573b..a43c87c1d343 100644
--- a/fs/cifs/transport.c
+++ b/fs/cifs/transport.c
@@ -194,8 +194,8 @@ smb_send_kvec(struct TCP_Server_Info *server, struct msghdr *smb_msg,
 
 	*sent = 0;
 
-	smb_msg->msg_name = (struct sockaddr *) &server->dstaddr;
-	smb_msg->msg_namelen = sizeof(struct sockaddr);
+	smb_msg->msg_name = NULL;
+	smb_msg->msg_namelen = 0;
 	smb_msg->msg_control = NULL;
 	smb_msg->msg_controllen = 0;
 	if (server->noblocksnd)
-- 
2.38.0


